src/librustc_mir/interpret/intern.rs

```diff
@@ -344,7 +365,10 @@ pub fn intern_const_alloc_recursive<M: CompileTimeMachine<'mir, 'tcx>>(
         } else if ecx.memory.dead_alloc_map.contains_key(&alloc_id) {
             // dangling pointer
             throw_unsup!(ValidationFailure("encountered dangling pointer in final constant".into()))
-        } else if ecx.tcx.alloc_map.lock().get(alloc_id).is_none() {
+        } else if let Some(_) = ecx.tcx.alloc_map.lock().get(alloc_id) {
+            // FIXME: check if the allocation is ok as per the interning rules as if we interned
+            // it right here.
```

Comment:
I don't understand this comment.

Reply:
If we encounter an `AllocId` that points to a mutable Allocation, (directly or via relocations in its `Allocation`), we should panic, the static rules should prevent this. We may hit an `AllocId` that belongs to an already interned static, and are thus not interning any further. But since we are also checking things during interning, we should probably continue doing those checks no matter what we encounter.

E.g. this should be unreachable for `InternKind::Promoted` except for allocations created for string and byte string literals.

Reply:
(This description ^--- should probably be added to the source...)